ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplLoadTask.java

```diff
@@ -499,6 +500,20 @@ private void dropTablesExcludedInReplScope(ReplScope replScope) throws HiveExcep
         dbName);
   }
 
+  private void createNotificationFileTask() {
+    if ((work.isIncrementalLoad() && !work.incrementalLoadTasksBuilder().hasMoreWork() && !work.hasBootstrapLoadTasks())
```

Comment:
Can simplify the condition:
!work.hasBootstrapLoadTasks() is required condition on both side of ||